internal/operands/tekton-pipelines/tekton-pipelines.go

```diff
@@ -0,0 +1,260 @@
+)
+
+// +kubebuilder:rbac:groups=tekton.dev,resources=pipelines,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=*,resources=configmaps,verbs=list;watch;create;delete
+// +kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=rolebindings,verbs=get;list;watch;create;update;patch;delete
+
+const (
+	namespacePattern = "^(openshift|kube)-"
+	operandName      = "tekton-pipelines"
+	operandComponent = common.AppComponentTektonPipelines
+)
+
+var namespaceRegex = regexp.MustCompile(namespacePattern)
+
+var requiredCRDs = []string{"tasks.tekton.dev"}
+
+func init() {
+	utilruntime.Must(pipeline.AddToScheme(common.Scheme))
+}
+
+type tektonPipelines struct {
+	pipelines       []pipeline.Pipeline
+	configMaps      []v1.ConfigMap
+	roleBindings    []rbac.RoleBinding
+	serviceAccounts []v1.ServiceAccount
+	clusterRoles    []rbac.ClusterRole
+}
+
+var _ operands.Operand = &tektonPipelines{}
+
+func New(bundle *tektonbundle.Bundle) *tektonPipelines {
+	tp := &tektonPipelines{
+		pipelines:       bundle.Pipelines,
+		configMaps:      bundle.ConfigMaps,
+		roleBindings:    bundle.RoleBindings,
+		serviceAccounts: bundle.ServiceAccounts,
+		clusterRoles:    bundle.ClusterRoles,
+	}
+	return tp
+}
+
+func (t *tektonPipelines) Name() string {
+	return operandName
+}
+
+func (t *tektonPipelines) WatchClusterTypes() []client.Object {
+	return []client.Object{
+		&pipeline.Pipeline{},
+		&v1.ConfigMap{},
+		&rbac.RoleBinding{},
+		&v1.ServiceAccount{},
+		&rbac.ClusterRole{},
+	}
+}
+
+func (t *tektonPipelines) WatchTypes() []client.Object {
+	return nil
+}
+
+func (t *tektonPipelines) RequiredCrds() []string {
+	return requiredCRDs
+}
+
+func (t *tektonPipelines) Reconcile(request *common.Request) ([]common.ReconcileResult, error) {
+	var results []common.ReconcileResult
+	var reconcileFunc []common.ReconcileFunc
+	reconcileFunc = append(reconcileFunc, reconcileClusterRolesFuncs(request, t.clusterRoles)...)
+	reconcileFunc = append(reconcileFunc, reconcileTektonPipelinesFuncs(t.pipelines)...)
+	reconcileFunc = append(reconcileFunc, reconcileConfigMapsFuncs(t.configMaps)...)
+	reconcileFunc = append(reconcileFunc, reconcileRoleBindingsFuncs(request, t.roleBindings)...)
+	reconcileFunc = append(reconcileFunc, reconcileServiceAccountsFuncs(request, t.serviceAccounts)...)
+
+	reconcileTektonBundleResults, err := common.CollectResourceStatus(request, reconcileFunc...)
+	if err != nil {
+		return nil, err
+	}
+
+	upgradingNow := isUpgradingNow(request)
+	for _, r := range reconcileTektonBundleResults {
+		if !upgradingNow && (r.OperationResult == common.OperationResultUpdated) {
+			request.Logger.Info(fmt.Sprintf("Changes reverted in tekton pipeline: %s", r.Resource.GetName()))
+		}
+	}
+	return append(results, reconcileTektonBundleResults...), nil
+}
+
+func (t *tektonPipelines) Cleanup(request *common.Request) ([]common.CleanupResult, error) {
+	var objects []client.Object
+	for _, p := range t.pipelines {
```

Comment:
These objects need to have the correct `namespace` before deleting them. Otherwise the operator may try to delete objects from a wrong namespace.

Reply:
Thanks!